apps/settings/js/modules/app_icon_helper.js

```diff
@@ -0,0 +1,49 @@
+define(function(require) {
+  'use strict';
+
+  // Return the url of the app icon that is the best match for the
+  // specified target size. This will return the URL of the smallest
+  // icon that is at least as big as the target. Or, if no such icon
+  // exists, it will return the largest icon. targetSize should be
+  // specified in CSS pixels. This function will convert to device
+  // pixels.
+  function getIconURL(app, targetSize) {
+    var manifest = app.manifest || app.updateManifest;
+    var icons = manifest && manifest.icons;
+
+    if (!icons || !Object.keys(icons).length) {
+      return '../style/images/default.png';
+    }
+
+    targetSize *= window.devicePixelRatio;
+    var bestSize = Infinity;
+    var maxSize = 0;
+
+    for (var size in icons) {
+      size = parseInt(size, 10);
+      if (size >= targetSize && size < bestSize) {
+        bestSize = size;
+      }
+      if (size > maxSize) {
+        maxSize = size;
+      }
+    }
+    // If there is an icon matching the preferred size, we return the result,
+    // if there isn't, we will return the maximum available size.
+    if (bestSize === Infinity) {
+      bestSize = maxSize;
+    }
+
+    var url = icons[bestSize];
+
+    if (url) {
+      return !(/^(http|https|data):/.test(url)) ? app.origin + url : url;
```

Comment:
Modifying to use URL(). Thanks for that. But if I use manifestURL instead of origin, then the icons for apps like Marketplace do not resolve.

Using manifestURL as the base URL, I get:  https://marketplace.firefox.com/app-icons/90.png
But using the origin, I get: app://packaged.marketplace.allizom.org/app-icons/90.png